paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/validation/PaymentContextValidator.java

```diff
@@ -0,0 +1,34 @@
+package fi.hel.verkkokauppa.payment.paytrail.validation;
+
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.common.error.Error;
+import fi.hel.verkkokauppa.payment.paytrail.context.PaytrailPaymentContext;
+import io.micrometer.core.instrument.util.StringUtils;
+import org.springframework.http.HttpStatus;
+
+public class PaymentContextValidator {
+
+    public static void validateContext(PaytrailPaymentContext context) {
+        if (context.isUseShopInShop()) {
```

Comment:
Lets try to keep code inside one if statement to keep code easier to be modified later and not so many if statements. 

First if statement should be `context.isUseShopInShop() && StringUtils.isEmpty(context.getShopId())` 
and returns first exception

Second if statement should be `!context.isUseShopInShop() && StringUtils.isEmpty(context.getPaytrailMerchantId()) || StringUtils.isEmpty(context.getPaytrailSecretKey())`
and return second exception